cmd/operator-sdk/cli/legacy.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/add"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/alpha"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/build"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/bundle"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/cleanup"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/completion"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/execentrypoint"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/generate"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/migrate"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/new"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/olm"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/printdeps"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/run"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/scorecard"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/test"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/version"
+	"github.com/operator-framework/operator-sdk/internal/flags"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func RunLegacy() error {
+	root := GetCLIRoot()
+	root.PersistentFlags().Bool(flags.VerboseOpt, false, "Enable verbose logging")
+	if err := viper.BindPFlags(root.PersistentFlags()); err != nil {
+		log.Fatalf("Failed to bind root flags: %v", err)
+	}
+
+	return root.Execute()
+}
+
+// GetCLIRoot is intended to create the base command structure for the OSDK for use in CLI and documentation
+// Deprecated: this is only used for projects without a PROJECT config/kubebuilder-style repo.
+func GetCLIRoot() *cobra.Command {
+	root := &cobra.Command{
+		Use:   "operator-sdk",
+		Short: "An SDK for building operators with ease",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			if viper.GetBool(flags.VerboseOpt) {
+				if err := projutil.SetGoVerbose(); err != nil {
+					log.Fatalf("Could not set GOFLAGS: (%v)", err)
+				}
+				log.SetLevel(log.DebugLevel)
+				log.Debug("Debug logging is set")
+			}
+			if err := checkGoModulesForCmd(cmd); err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+
+	root.AddCommand(
+		add.NewCmd(),
+		alpha.NewCmd(),
+		build.NewCmd(),
+		bundle.NewCmd(),
+		cleanup.NewCmd(),
+		completion.NewCmd(),
+		execentrypoint.NewCmd(),
+		generate.NewCmd(),
+		migrate.NewCmd(),
+		new.NewCmd(),
+		olm.NewCmd(),
+		printdeps.NewCmd(),
+		run.NewCmd(),
+		scorecard.NewCmd(),
+		test.NewCmd(),
+		version.NewCmd(),
+	)
+
+	// The "init" cmd is is added as a hidden subcommand from the new KB CLI to provide
+	// a way to initialize Kubebuilder projects from the legacy CLI.
+	// TODO(hasbro17): This should be removed once the KB CLI is the default.
+	initCmd := getInitCmd()
```

Comment:
Init with the domain is not working. 

```
$ operator-sdk init --domain my.domain
Error: failed to initialize project with version "3-alpha": error finding current repository: could not determine repository path from module data, package data, or by initializing a module: go: cannot determine module path for source directory /Users/camilamacedo/go/src (outside GOPATH, module path must be specified)

Example usage:
	'go mod init example.com/m' to initialize a v0 or v1 module
	'go mod init example.com/m/v2' to initialize a v2 module

Run 'go help mod init' for more information.

Usage:
  operator-sdk init [flags]

Examples:
  # Scaffold a project using the apache2 license with "The Kubernetes authors" as owners
  operator-sdk init --project-version=2 --domain example.org --license apache2 --owner "The Kubernetes authors"


Flags:
      --domain string            domain for groups (default "my.domain")
      --fetch-deps               ensure dependencies are downloaded (default true)
  -h, --help                     help for init
      --license string           license to use to boilerplate, may be one of 'apache2', 'none' (default "apache2")
      --owner string             owner to add to the copyright
      --project-version string   project version, possible values: ("2", "3-alpha") (default "3-alpha")
      --repo string              name to use for go module (e.g., github.com/user/repo), defaults to the go package of the current working directory.
      --skip-go-version-check    if specified, skip checking the Go version

Global Flags:
      --verbose   Enable verbose logging
```

c/c @estroz @hasbro17

Reply:
Where are you running your command and what go version are you using?

Reply:
@camilamacedo86 Did you run `go mode init` before `operator-sdk init`?
Or use the `--repo` flag?

Reply:
Let's close this one .. I am unable to see the error again. I probably did something wrong.